lib/algolia_index/models/firm.rb

```diff
@@ -0,0 +1,15 @@
+module AlgoliaIndex
+  class Firm < Base
+    def update!
+      advisers = object&.advisers&.geocoded
+      return if advisers.blank?
+
+      serialized = advisers.map(&AlgoliaIndex::AdviserSerializer.method(:new))
+      AlgoliaIndex.index_advisers.add_objects(serialized)
+    end
+
+    def destroy!
+      # Nothing to be directly destroyed in any index
```

Comment:
Worth raising an exception of some sort then, rather than just returning nil?

Reply:
This is a very very good comment. Although there are some objects that call then and in some should be doing something and others not. Based on this comment, me and @your are thinking to rename to just `destroy` (without a bang) so that the return nil make sense. Wdyt?

Reply:
@malcolmbaig That's a very good comment, thanks for raising it.

Since this method is triggered anyway when a firm is destroy, it should not do anything, as we don't have a firm index.

But whenever a firm is destroyed in the db, their advisers and offices will be destroyed because of the cascade, which will trigger the removal of advisers and offices from their relative indices. 

So this method here, should not do anything really, but it's very confusing to have the banged version of it when it does not raise indeed, so I think we could just remove the bang and avoid confusion.

What do you think, does it make sense?

Reply:
I think that makes sense - I'd take it a step further and remove the bangs from other methods as well. After removing from destroy, we no longer have a consistent meaning for `!` across these objects so I don't think it adds clarity, IMHO. 

A slightly more detailed comment would help too I reckon, perhaps something like: 

`# Do nothing, because we don't maintain a Firm index.`

Reply:
@malcolmbaig Totally, needs to be removed from all of them. Will rephrase the comment if that's clearer. 👍